quic/s2n-quic-transport/src/space/application.rs

```diff
@@ -539,6 +544,50 @@ impl<Config: endpoint::Config> ApplicationSpace<Config> {
 
         limits
     }
+
+    // Store ACKs in PendingAckRanges for delayed processing
+    pub fn update_pending_acks<A: frame::ack::AckRanges>(
+        &mut self,
+        frame: &frame::Ack<A>,
+        pending_ack_ranges: &mut PendingAckRanges,
+    ) -> Result<(), transport::Error> {
+        let range = frame.ack_ranges().into_iter().map(|f| {
+            PacketNumberRange::new(
+                PacketNumberSpace::ApplicationData.new_packet_number(*f.start()),
+                PacketNumberSpace::ApplicationData.new_packet_number(*f.end()),
+            )
+        });
+        pending_ack_ranges
+            .extend(range, frame.ecn_counts, frame.ack_delay())
+            .or(
+                // TODO: post metrics: ack ranges were dropped
+                Ok(()),
+            )
```

Comment:
If you can't fit anymore ack ranges in the pending_ack_ranges, wouldn't you want to just process the rest of the acks immediately or process all the pending acks immediately?

Reply:
Good point/question.

Reply:
enabling ack delay and the exact implementation will happen in a follow up PR but i have captured your comment here: https://github.com/aws/s2n-quic/pull/1293/files#diff-d9b11e2280bc4fe94160b8543a089d4c7fe0fb5b037d9c1fc491e1304e757d45R752-R755